File: Go/Find-The-Index-Of-The-First-Occurrence-In-A-String.go
package main

func strStr(haystack string, needle string) int {
	//  strings.Index(haystack, needle)
	lenNeedle := len(needle)
	firstLetter := needle[0]

	for i := 0; i <= len(haystack)-lenNeedle; i++ {
		if haystack[i] == firstLetter && haystack[i:i+lenNeedle] == needle {
			return i
		}
	}

	return -1
}

File: Javascript/Find-The-Index-Of-The-First-Occurrence-In-A-String.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function (haystack, needle) {
  const lenNeedle = needle.length;
  const firstLetter = needle[0];

  for (let i = 0; i <= haystack.length - lenNeedle; i++) {
    if (
      haystack[i] == firstLetter &&
      haystack.slice(i, i + lenNeedle) == needle
    ) {
      return i;
    }
  }

  return -1;
  //   return haystack.indexOf(needle);
};
